VMEncryption/main/oscrypto/rhel_72/encryptstates/UnmountOldrootState.py

```diff
@@ -112,12 +118,37 @@ def enter(self):
 
             self.command_executor.Execute('kill -9 {0}'.format(victim))
 
+        # Re-execute systemd, get pid 1 to use the new root
         self.command_executor.Execute('telinit u', True)
 
         sleep(3)
 
         self.command_executor.Execute('umount /oldroot', True)
 
+        self.restart_systemd_services()
+        proc_comm = ProcessCommunicator()
+        self.command_executor.ExecuteInBash(
+                command_to_execute="grep {0} /proc/*/task/*/mountinfo".format(self.rootfs_block_device),
+                raise_exception_on_failure=True,
+                communicator=proc_comm)
+
+        procs_to_kill = filter(lambda path: path.startswith('/proc/'), proc_comm.stdout.split())
+        procs_to_kill = map(lambda path: int(path.split('/')[2]), procs_to_kill)
+        procs_to_kill = reversed(sorted(procs_to_kill))
+        self.context.logger.log("Processes with tasks using {0}:\n{1}".format(self.rootfs_block_device, procs_to_kill))
```

Comment:
How well does this logging statement work? In python 3, the reversed and sorted functions will return lazy objects. Printing them will result in something like: <list_reverseiterator object at 0x000001F987BAA8D0>

Reply:
Also, this whole chunk took a while to understand. Some comments explaining how the "procs_to_kill" are selected would be helpful here.

Reply:
Maybe add a list() around the last reversed() to force execution then...